### PR TITLE
added support for shippable

### PR DESCRIPTION
**Pull request by caarlos0**
I wasn't able to test it in my machine though (problems with node-gyp)...

Anyways, I think it should work that way..

### CODE REVIEW

**Comment by espadrine** [comment]
You do need a couple of installed libraries to run shields: phantomjs and cairo (see [here](https://github.com/badges/shields/blob/master/INSTALL.md#start-the-server)).

It would be preferable to avoid SVG scraping. Do you know if there are plans to support this in Shippable's standard API?

Your PR as it stands triggers an Internal Server Error in shippable's server.

File: server.js [hunk 1]
```diff
@@ -312,6 +312,41 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Shippable integration
+camp.route(/^\/shippable?\/([^\/]+)(?:\/(.+))?\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var project = match[2];  // eg, 54d119db5ab6cc13528ab183
+  var branch = match[3];
+  var format = match[4];
```
Comment (espadrine) [inline server.js]: 2, 3, 4? I think 1, 2, 3.

**Comment by espadrine** [comment]
Thanks for contributing anywho!

File: server.js (hunk 1 above)
Comment (caarlos0) [inline server.js]: yeah, sorry..

**Comment by caarlos0** [comment]
@espadrine thing is: for some reason I can't install cairo here.. got different errors every time...

Don't know about the shippable api..

Thanks!

**Comment by mahirk** [comment]
@espadrine @caarlos0  it seems like this service no longer works, it requires adding a whitespace replace to `fetchFromSvg`
```
var badge = buffer.replace(/(?:\r\n\s|\r\s|\n\s)/g, '');
var match = />([^<>]+)<\/text><\/g>/.exec(badge);
```
Sending in a PR for the same

**Comment by caarlos0** [comment]
@mahirk yeah, maybe they changed the API...

File: server.js [hunk 2]
```diff
@@ -312,6 +312,41 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Shippable integration
+camp.route(/^\/shippable?\/([^\/]+)(?:\/(.+))?\.(svg|png|gif|jpg|json)$/,
```
Comment (espadrine) [inline server.js]: The first `?` seems unnecessary.

**Pull request merged**